src/core/public/legacy_platform/legacy_platform_service.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { InjectedMetadataStartContract } from '../injected_metadata';
+
+interface Deps {
+  injectedMetadata: InjectedMetadataStartContract;
+}
+
+export interface LegacyPlatformParams {
+  rootDomElement: HTMLElement;
+  requireLegacyFiles: () => void;
+  useLegacyTestHarness?: boolean;
+}
+
+export class LegacyPlatformService {
```

Comment:
__suggestion:__ same note about top level comment here as well. We can add these comments in the following PRs if you want though.